lib/data/entity.js

```diff
@@ -417,6 +475,26 @@ const getWithConflictDetails = (defs, audits, relevantToConflict) => {
 
   const relevantBaseVersions = new Set();
 
+  // build up branches
+  const branches = new Map();
+  for (const version of defs) {
+    const { branchId } = version;
+    if (branchId != null && version.branchBaseVersion != null) {
+      const existingBranch = branches.get(branchId);
+      if (existingBranch == null) {
+        const newBranch = new Branch(version, defs[0]);
+        branches.set(branchId, newBranch);
+        version.branch = newBranch;
```

Comment:
I know we do it in Frontend, but I think we should avoid mutating `version` here unless it's necessary.